virtcontainers: Add automatic crashdump collection for Cloud  Hypervisor

Ankita13-code opened the pull request:
###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
<!-- **All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)* -->
- [ ] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
- [ ] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [ ] Merged using "create a merge commit" rather than "squash and merge" (or similar)
- [ ] genPolicy only: Builds on Windows
- [ ] genPolicy only: Updated sample YAMLs' policy annotations, if applicable

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of the changes. -->
Implement automatic guest memory dump collection when a guest VM panics in Cloud Hypervisor, achieving feature parity with QEMU hypervisor.

Implementation:
- Monitor CLH event file for panic events with non-blocking I/O
- Enable pvpanic device and configure event-monitor when crashdump enabled
- Set `panic=0` kernel cmdline param to prevent reboot during memory dump
- Dump guest memory to ELF format with hypervisor metadata
- Add `guest_memory_dump_path` option to configuration-clh.toml.in

Memory dumps saved to <guest_memory_dump_path>/<sandbox-id>/ including vmcore ELF file, hypervisor config/version, and sandbox state.

Requires CLH built with `guest_debug` feature.

This enables automated crash analysis workflows for Kata Containers with Cloud Hypervisor, similar to existing QEMU functionality.
###### Associated issues  <!-- optional -->
<!-- Link to Github issues if possible. -->

###### Test Methodology
<!-- How was this test validated? i.e. local build, pipeline build etc. -->

Ankita13-code committed:
virtcontainers: Add automatic crashdump collection for Cloud Hypervisor

Implement automatic guest memory dump collection when a guest VM panics
in Cloud Hypervisor, achieving feature parity with QEMU hypervisor.

Implementation:
- Monitor CLH event socket for panic events with non-blocking I/O
- Enable pvpanic device and configure event-monitor when crashdump enabled
- Set panic=-1 kernel param to prevent reboot during memory dump
- Dump guest memory to ELF format with hypervisor metadata
- Add guest_memory_dump_path option to configuration-clh.toml.in

Memory dumps saved to <guest_memory_dump_path>/<sandbox-id>/ including
vmcore ELF file, hypervisor config/version, and sandbox state.

Requires CLH built with guest_debug feature.

This enables automated crash analysis workflows for Kata Containers with
Cloud Hypervisor, similar to existing QEMU functionality.

Signed-off-by: Ankita Pareek <ankitapareek@microsoft.com>

sprt commented on src/runtime/virtcontainers/clh.go:
```diff
@@ -1457,6 +1519,259 @@ func (clh *cloudHypervisor) launchClh() error {
+// waitForEventMonitorFile waits for the event monitor file to be created by CLH
+func (clh *cloudHypervisor) waitForEventMonitorFile(eventMonitorPath string, timeout time.Duration) error {
+	timeoutChan := time.After(timeout)
+	ticker := time.NewTicker(10 * time.Millisecond)
```
I wouldn't check more often than every 100ms?
```suggestion
	ticker := time.NewTicker(100 * time.Millisecond)
```

sprt commented on src/runtime/virtcontainers/clh.go:
```diff
@@ -1457,6 +1519,259 @@ func (clh *cloudHypervisor) launchClh() error {
+			clh.Logger().Info("CLH event monitoring stopped via stop channel")
+			return
+		case event := <-watcher.Events:
+			if event.Op&fsnotify.Write == fsnotify.Write {
```
I think the lack of space around `&` is per gofmt.

Redent0r commented:
@claude create a markdown file summarizing these changes
include:
- high level summary
- major changes
- breaking changes
- anything Azure-specific

Copilot committed:
docs: add crashdump feature summary for Cloud Hypervisor

Agent-Logs-Url: https://github.com/microsoft/kata-containers/sessions/5688eec1-3de7-45b6-b96b-b04baf93df78

Co-authored-by: Redent0r <30801614+Redent0r@users.noreply.github.com>
